kernel/src/process_standard.rs

```diff
@@ -1307,12 +1298,18 @@ impl<C: 'static + Chip> ProcessStandard<'_, C> {
     const PROCESS_STRUCT_OFFSET: usize = mem::size_of::<ProcessStandard<C>>();
 
     /// Create a `ProcessStandard` object based on the found `ProcessBinary`.
+    ///
+    /// The `storage_permissions_policy` is optional because any app without a
+    /// fixed `ShortId` cannot have storage permissions, so in cases where an
+    /// AppID assigner is not used it doesn't make sense to have a storage
+    /// permissions policy.
```

Comment:
Don't we have the null-policy for that?

Reply:
That is an option. However, we would still need a `&'static ProcessStandardStoragePermissionsPolicy` which is, I think, a fair bit of confusing overhead for something that doesn't do anything.

Reply:
Rust allows us to create `&'static` references to zero-sized types out of thin air: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2b0c391057f50adfb07c62608ff1ef0f

Alternatively, as it's slightly weird to return references from a constructor, we can also make the constructor `const` and then store the result in a static (which, as it's zero-sized, will not have any runtime overhead): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9c1f4334f81afdbd6a43d39055de7f70
In particular, this doesn't seem too bad of an option as the null policy doesn't require a capability to construct it.

I don't like that we're using an `Option` here, as that's one additional branch point for something that we already have semantically encapsulated with the null policy.

Reply:
Ok! Implementing the null permissions for `()` worked and the `Option<Policy>` is gone.